### ISSUE TITLE
Parallel processing for downloading and unziping

### ISSUE THREAD

**Issue by koleckt** [CS-SI/eodag#146]
We can save a lot of processing time, specially for large dataset of product (time serie) if the downloading and the unziping (exemple of S1 products) can be done in 2 differents threads.

Actually, when downloading, the CPU is not used at full capacity, and when unzip, the network is not use.

Doing both in parallel can speedup a lot the processing codes.

**Comment by sbrunato** [comment]
Thanks for this suggestion, we will work on a new feature that provide a way to parallelize products downloading, and keep you informed of the progress in this issue

**Comment by sbrunato** [comment]
Related to #120

**Comment by sbrunato** [comment]
#120

**Comment by sbrunato** [comment]
See if there is a way to provide to the downloading method a parameter referencing to a generic pool / worker scheduler, compatible with Dask or other solutions

Ask as parameter an executor like `Dask.Client`, or `concurrent.futures` that use the same interfaces. See https://distributed.dask.org/en/latest/client.html

Some serializing might be needed (already implemented in eodag):
- see GH-140 (closed)

**Comment by alambare** [comment]
We will implement this feature using a `ThreadpoolExecutor` to follow the logic of what is already done in the library.
The method should also expose a `executor` parameter to allow users to provide their own if needed.

**Comment by sbrunato** [comment]
Also use the `executor` in `download()` to download multiple assets in parallel (create one if non is given).
When using `download_all()`, internally pass the `executor` to `download()`
